tests/test_butler.py

```diff
@@ -619,6 +619,13 @@ def setUp(self):
         uri = ButlerURI(config[".datastore.datastore.root"])
         self.bucketName = uri.netloc
 
+        # set up some fake credentials if they do not exist
+        if not os.path.exists("~/.aws/credentials"):
+            if "AWS_ACCESS_KEY_ID" not in os.environ:
+                os.environ["AWS_ACCESS_KEY_ID"] = "dummyAccessKeyId"
+            if "AWS_SECRET_ACCESS_KEY" not in os.environ:
```

Comment:
If only one of these environment variables has been set is that a reasonable scenario? Does it make sense to have one be real and one be dummy?  If it's all or nothing then I suggest you check for existence of both env vars and don't set dummy values if both are already set.

Reply:
I don't believe so. Conditional changed.